content/chainguard/chainguard-images/using-the-tag-history-api.md

```diff
@@ -0,0 +1,111 @@
+```
+
+Where `IMAGE_NAME` is the name of the image that you want to pull the tag history from. It's worth noting that the token is only valid for pulling the history of that specific image.
+
+For public images (tagged as `latest` or `latest-dev`), you can request a registry token anonymously, without providing any pre-existing auth.
+
+The following command will obtain a token for the **Python** image and register a variable called `tok` with the resulting value, which you can use in a subsequent command to obtain the tag history:
+
+```shell
+tok=$(curl "https://cgr.dev/token?scope=repository:chainguard/python:pull" | jq -r .token)
+```
+
+For images that are not part of the public catalog, you'll need to exchange your Chainguard token for a registry token. This assumes you've set up auth with [chainctl auth configure-docker](https://edu.chainguard.dev/chainguard/chainguard-images/registry/authenticating/):
+
+```shell
+tok=$(curl -H "Authorization: Bearer $(echo 'cgr.dev' | docker-credential-cgr get)" -v "https://cgr.dev/token?scope=repository:chainguard/python:pull" | jq -r .token)
+```
+
+To make sure your token is set, you can run the following command:
+
+```shell
+echo $tok
+```
+
+And you should get a long string token as output.
+
+You should now be ready for calling the API, either manually or programmatically.
+
+## Calling the API
+
+Once you have your token available, you can run a `curl` query passing on your token within an `Authorization: bearer` header to the following endpoint:
+
+```
+https://cgr.dev/v2/chainguard/IMAGE_NAME/_chainguard/history/IMAGE_TAG
+```
+
+Where `IMAGE_NAME` is the name of the image, for instance: `python`, and `IMAGE_TAG` is the tag that you want to pull history from.
+
+For example, this is how you can fetch the tag history of the **python:latest** Chainguard image using `curl` on the command line:
+
+```shell
+curl -H "Authorization: Bearer $tok" https://cgr.dev/v2/chainguard/python/_chainguard/history/latest | jq
+```
+
+You should get output like the following:
+
+```
+{
+  "history": [
+    {
+      "updateTimestamp": "2023-05-12T13:46:10.555Z",
+      "digest": "sha256:81c334de6dd4583897f9e8d0691cbb75ad41613474360740824d8a7fa6a8fecb"
+    },
+    {
+      "updateTimestamp": "2023-05-12T20:50:19.702Z",
+      "digest": "sha256:a8724b7a80cae14263a3b55f7acb5d195fcbb24afbc8067aa5198aa2a9131cde"
+    },
+    ...
+
+  ]
+}
+```
+
+## Using Image Digests within a Dockerfile
+
+Setting up your Dockerfile to use a stale build is a matter of modifying your `FROM` line to use an image digest instead of a tag. For instance, let's say you want to make sure you keep using the current latest build of the Python image. In a previous section of this page we obtained the tag history of the Python image, and the most recent build digest is listed as `sha256:81c334de6dd4583897f9e8d0691cbb75ad41613474360740824d8a7fa6a8fecb`. With that information, you can edit your Dockerfile and replace:
```

Comment:
```suggestion
Setting up your Dockerfile to use an older build is a matter of modifying your `FROM` line to use an image digest instead of a tag. For instance, let's say you want to make sure you keep using the current latest build of the Python image. In a previous section of this page we obtained the tag history of the Python image, and the most recent build digest is listed as `sha256:81c334de6dd4583897f9e8d0691cbb75ad41613474360740824d8a7fa6a8fecb`. With that information, you can edit your Dockerfile and replace:
```